reactor-kafka-samples/src/main/java/reactor/kafka/samples/SampleScenarios.java

```diff
@@ -174,7 +174,7 @@ public KafkaSource(String bootstrapServers, String topic) {
                            .receive()
                            .publishOn(scheduler)
                            .concatMap(m -> storeInDB(m.value())
-                                          .doOnSuccess(r -> m.receiverOffset().commit().block()))
+                                          .delayUntil(r -> m.receiverOffset().commit()))
```

Comment:
Since `.delayUntil` is implemented as `it -> .concatMap(...).thenReturn(it)`, I would suggest using `.concatMap` here for the same of performance. The result is `Flux<?>`, should be okay

Reply:
done